frontend/app/src/components/status/UpdateIndicator.vue

```diff
@@ -10,19 +10,37 @@
 </template>
 
 <script lang="ts">
-import { Component, Vue } from 'vue-property-decorator';
+import { Component, Vue, Watch } from 'vue-property-decorator';
 import { mapGetters, mapState } from 'vuex';
 import { Version } from '@/store/types';
 
 @Component({
   computed: {
     ...mapState(['version']),
+    ...mapGetters('settings', ['versionUpdateCheckFrequency']),
     ...mapGetters(['updateNeeded'])
   }
 })
 export default class UpdateIndicator extends Vue {
   updateNeeded!: boolean;
   version!: Version;
+  versionUpdateCheckFrequency!: number;
+  refreshInterval: any;
+
+  @Watch('versionUpdateCheckFrequency')
```

Comment:
Replace with this for testing, it should check every 5 seconds, with 50% probability that there are update.
`@Watch('versionUpdateCheckFrequency')
async setVersionUpdateCheckInterval() {
  this.refreshInterval = setInterval(async () => {
    await this.$store.dispatch('version');
  }, 5000);
}`

Reply:
And also for this file `frontend/app/src/store/store.ts`, you can change the `version` action into these:

```
async version({ commit }): Promise<void> {
  const { version, dataDirectory } = await api.info();
  if (Math.random() > 0.5) {
    const mockedVersion: BackendVersion = {
      ourVersion: '10',
      latestVersion: '12',
      downloadUrl: 'https://google.com'
    };
    commit('versions', mockedVersion);
  } else if (version) {
    commit('versions', version);
  }

  commit('dataDirectory', dataDirectory);
},
```